test/cli/acceptance.go

```diff
@@ -457,27 +457,40 @@ func Version() (string, error) {
 
 // AddAlertsAt declares alerts that are to be added to the Alertmanager
 // server at a relative point in time.
-func (am *Alertmanager) AddAlertsAt(at float64, alerts ...*TestAlert) {
+func (am *Alertmanager) AddAlertsAt(useHeuristic bool, at float64, alerts ...*TestAlert) {
 	am.t.Do(at, func() {
-		am.AddAlerts(alerts...)
+		am.AddAlerts(useHeuristic, alerts...)
 	})
 }
 
 // AddAlerts declares alerts that are to be added to the Alertmanager server.
-func (am *Alertmanager) AddAlerts(alerts ...*TestAlert) {
+func (am *Alertmanager) AddAlerts(useHeuristic bool, alerts ...*TestAlert) {
 	for _, alert := range alerts {
-		out, err := am.addAlertCommand(alert)
+		out, err := am.addAlertCommand(useHeuristic, alert)
 		if err != nil {
 			am.t.Errorf("Error adding alert: %v\nOutput: %s", err, string(out))
 		}
 	}
 }
 
-func (am *Alertmanager) addAlertCommand(alert *TestAlert) ([]byte, error) {
+func (am *Alertmanager) addAlertCommand(useHeuristic bool, alert *TestAlert) ([]byte, error) {
 	amURLFlag := "--alertmanager.url=" + am.getURL("/")
 	args := []string{amURLFlag, "alert", "add"}
-	for key, val := range alert.labels {
-		args = append(args, key+"="+val)
+	// Make a copy of the labels
+	labels := make(models.LabelSet, len(alert.labels))
+	for k, v := range alert.labels {
+		labels[k] = v
+	}
+	if useHeuristic {
+		// If alertname is present and useHeuristic is true then the command should
+		// be `amtool alert add foo ...` and not `amtool alert add alertname=foo ...`.
```

Comment:
The difference is that some tests like `TestAddAlert` for example want to test both variants of creating an alert in amtool:

```
amtool alert add foo
```

```
amtool alert add alertname=foo
```

Not all tests that create alerts need to test both variants and some of those tests just create alerts as part of their test fixture, but for other tests like `TestAddAlert` running these variants is the test itself.

This change was added to avoid having to rewrite how alerts are created for all tests when the use case is just `TestAddAlert`.

---

Something like that?

Reply:
A summary of that sounds good.

I think the most important parts are:

> some tests want to test both variants of creating an alert in amtool:

> This change was added to avoid having to rewrite how alerts are created for all tests when the use case is jus